api/payments.js

```diff
@@ -122,19 +123,19 @@ function formatPaymentHelper(account, transaction, callback) {
  *          submitted transaction
  */
 function submitPayment(account, payment, clientResourceID, secret,
-    lastLedgerSequence, urlBase, options, callback) {
+    urlBase, options, callback) {
   var self = this;
   var max_fee = Number(options.max_fee) > 0 ?
     xrpToDrops(options.max_fee) : undefined;
   var fixed_fee = Number(options.fixed_fee) > 0 ?
     xrpToDrops(options.fixed_fee) : undefined;
 
-  validate.client_resource_id(clientResourceID);
-  // TODO: validate.addressAndSecret({address: account, secret: secret});
-  validate.address(account);
+  // these checks are done at the top of the function because we cannot
+  // raise exceptions inside callbacks
```

Comment:
This is because we are catching the exceptions using the express middleware, right?